art_internal_impl.hpp

```diff
@@ -196,13 +226,18 @@ inline void basic_db_inode_deleter<INode, Db>::operator()(
 #endif  // UNODB_DETAIL_WITH_STATS
 }
 
-template <class Db, template <class> class CriticalSectionPolicy, class NodePtr,
+// The basic_art_policy encapsulates differences between plain and OLC
+// ART, such as extra header field, and node access critical section
+// type.
+template <class Db, template <class> class CriticalSectionPolicy,
+          class LockPolicy, class ReadCriticalSection, class NodePtr,
```

Comment:
Passes the LockPolicy and ReadCriticalSection policy to support faking those things for the non-OLC implementations.